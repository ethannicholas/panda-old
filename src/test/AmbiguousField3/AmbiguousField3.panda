================================================================================
The reference to "t.length" is ambiguous because the class has both a field 
and a no-arg function by that name. Outside of the class with both field and
function visible, we should complain about ambiguity.
================================================================================
class AmbiguousField3 {
    var length:Int := 5

    function length():Int {
        return length
    }
}

method main() {
    var t := new AmbiguousField3()
    Console.writeLine(t.length)
}