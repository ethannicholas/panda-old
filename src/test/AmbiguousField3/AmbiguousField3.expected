error: AmbiguousField3.panda:16:25: reference to 'length' is ambiguous, both field length (of type panda.core.Int32) and function length() match
1 error
